implementations/php-json-schema/bowtieJsonSchema.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+require_once 'vendor/autoload.php';
+
+use Opis\JsonSchema\{
+    Validator
+};
+
+$STARTED = false;
+
+function start($request)
+{
+    if ($request->version !== 1) {
+        throw new Exception('Wrong version!');
+    }
+
+    $GLOBALS['STARTED'] = true;
+
+    $response = [
+        'ready' => true,
+        'version' => 1,
+        'implementation' => [
+            'language' => 'php',
+            'name' => 'json-schema',
+            'homepage' => 'https://opis.io/json-schema',
+            'issues' => 'https://github.com/opis/json-schema/issues',
+            'dialects' => [
+                'https://json-schema.org/draft/2020-12/schema',
+                'https://json-schema.org/draft-2019-09/schema',
+                'http://json-schema.org/draft-07/schema#',
+                'http://json-schema.org/draft-06/schema#',
+            ],
+            'os' => PHP_OS,
+            'os_version' => php_uname('r'),
+            'language_version' => PHP_VERSION,
+        ],
+    ];
+    return json_encode($response, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+}
+
+function dialect($request)
+{
+    if (!$GLOBALS['STARTED']) {
+        throw new Exception('Not started!');
+    }
+    return json_encode(['ok' => false], JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+}
+
+function run($request)
+{
+    $validator = new Validator();
+    $resolver = $validator->loader()->resolver();
+
+    if (!$GLOBALS['STARTED']) {
+        throw new Exception('Not started!');
+    }
+
+    if (isset($request->case->registry)) {
+        foreach ($request->case->registry as $key => $value) {
+            $resolver->registerRaw($value, $key);
+        }
+    }
+
+    $results = [];
+
+    foreach ($request->case->tests as $test) {
+        $result = $validator->validate($test->instance, $request->case->schema);
+        $results[] = ['valid' => $result->isValid()];
+    }
+    $response = ['seq' => $request->seq, 'results' => $results];
+    return json_encode($response, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+}
+
+function stop($request)
+{
+    if (!$GLOBALS['STARTED']) {
+        throw new Exception('Not started!');
+    }
+
+    exit(0);
+}
+
+$cmds = [
+    'start' => 'start',
+    'dialect' => 'dialect',
+    'run' => 'run',
+    'stop' => 'stop'
+];
+
+while (($line = fgets(STDIN)) !== false) {
+    $request = json_decode($line);
+    if (json_last_error() !== JSON_ERROR_NONE) {
+        echo json_encode(['error' => json_last_error_msg()], JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE) . "\n";
```

Comment:
This doesn't look right, technically -- what you have basically will print a JSON object `{"error": "foowhatevermessage"}` whenever it runs into a decode error.

But that might violate Bowtie's protocol.

Bowtie does not always allow you to return an object like that.

That's only for `run` requests, where the key is called `errored`, see https://github.com/bowtie-json-schema/bowtie/blob/main/bowtie/schemas/io-schema.json#L212-L235

But you can't simply return some other object whenever an error occurs, you're supposed to follow Bowtie's protocol, and for example a `start` request, if that fails for whatever reason, there is no `error` message response.

If you *do* want to catch errors here, and I'm sure you do as you said this is where your initial issue was, then the way you can report those is by writing them to stderr -- and then that doesn't have to be JSON.

Specifically it sounds like for whatever reason that PHP is/was writing exceptions to stdout -- so you should check how to configure it to write them to stderr, and then you may not need to do any extra work here, bowtie will notice errors on stderr and continue.